initrd/init

```diff
@@ -337,19 +294,62 @@ for c in `cat /proc/cmdline`; do
 done
 
 cd /
-while :; do
-	for device in ${ROOT:-/dev/[hmnsvx][dmrv][0-9a-z]*}; do
-		check_root $device && break 2
+for device in /dev/[hmnsvx][dmrv][0-9a-z]*; do
+	ln $device /dev/block
+done
 
-		# Also umount /android and /sfs if already mounted
-		mountpoint -q /android && umount /android
-		mountpoint -q /sfs && umount /sfs
-		mountpoint -q /mnt && umount /mnt
-	done
-	sleep 1
-	echo -n .
+# if the android is partition table-based
+IS_ALL_BLOCKDEV=1
+for dev in $SYSTEM $DATA $RECOVERY; do
+	case "$dev" in
+	/dev/*) ;;
+	*) IS_ALL_BLOCKDEV= && break ;;
+	esac
 done
 
+if [ -z "$IS_ALL_BLOCKDEV" ]; then
+
+	ROOT=$(find_device "$ROOT")
+	if [ -z "$ROOT" ]; then
+		source /hooks/0-find-rootdev
+		find_rootdev 10 || debug_shell debug-notfound # maximum 10 iterations
+	else
+		try_mount ro $dev /mnt || debug_shell debug-mountrootfail
+	fi
+
+	if [ -n "$INSTALL" ]; then
+		if [ ! -d "/mnt/isolinux" ] || [ ! -f "/mnt/install.img" ]; then
+			debug_shell debug-missinginstall
```

Comment:
What's `debug-missinginstall` ?

Reply:
At first these 5 lines was in `check_root` function but as I modified it these got moved out of the function
It checks if INSTALL is specified but whether install image exist or not. If not, it launches the interactive shell for user to manually fix the problem

Reply:
but then there should be a `debug-missinginstall`  function somewhere right ? I don't see it anywhere, same with `debug-mountsystemfail`

Reply:
Debug message; not implemented

Reply:
either implement or remove it